### PR TITLE
add analyzer excution sidecar

### PR DESCRIPTION
**Pull request by jgrillo-grapl**
### Which issue does this PR correspond to?
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/1013

### What changes does this PR make to Grapl? Why?
This PR adds an analyzer execution sidecar (more or less like the generator execution sidecar)

### How were these changes tested?
kinda tbd, automated tests

### CODE REVIEW

File: src/rust/plugin-execution-sidecar/analyzer-execution-sidecar.rs [hunk 1]
```diff
@@ -1,4 +1,28 @@
+use clap::Parser;
+use grapl_tracing::setup_tracing;
+use plugin_execution_sidecar::{
+    config::PluginExecutorConfig,
+    plugin_executor::PluginExecutor,
+    work::AnalyzerWorkProcessor,
+};
+
+const SERVICE_NAME: &'static str = "generator-execution-sidecar";
```
Comment (wimax-grapl) [inline src/rust/plugin-execution-sidecar/analyzer-execution-sidecar.rs:9]: alert alert

File: src/rust/plugin-execution-sidecar/src/work/analyzer_work_processor.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+pub struct AnalyzerWorkProcessor {
+    analyzer_service_client: AnalyzerServiceClient,
+}
+
+impl AnalyzerWorkProcessor {
+    pub async fn new(config: &PluginExecutorConfig) -> Result<Self, Box<dyn std::error::Error>> {
+        let plugin_id = config.plugin_id;
+        let upstream_addr_env_var = format!("NOMAD_UPSTREAM_ADDR_plugin-{plugin_id}");
+        let upstream_addr = std::env::var(&upstream_addr_env_var).expect(&upstream_addr_env_var);
+        let address = format!("http://{upstream_addr}");
```
Comment (colin-grapl) [inline src/rust/plugin-execution-sidecar/src/work/analyzer_work_processor.rs:56]: Can we get a debug/info log here?

**Review by colin-grapl — APPROVED** [review]

**Review by wimax-grapl — APPROVED** [review]

File: src/proto/graplinc/grapl/api/plugin_sdk/analyzers/v1beta1/analyzers.proto [hunk 3]
```diff
@@ -96,8 +96,6 @@ message ExecutionMiss {
 
 // A request to execute an Analyzer against a specific update to a tenant's graph
 message RunAnalyzerRequest {
-  // The tenant that the graph belongs to
-  graplinc.common.v1beta1.Uuid tenant_id = 1;
   // The update that occurred
   Update update = 2;
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/proto/graplinc/grapl/api/plugin_sdk/analyzers/v1beta1/analyzers.proto]: fixed

File: src/rust/plugin-execution-sidecar/src/work/analyzer_work_processor.rs (hunk 2 above)
Comment (jgrillo-grapl) [reply src/rust/plugin-execution-sidecar/src/work/analyzer_work_processor.rs:56]: fixed

File: src/rust/plugin-execution-sidecar/analyzer-execution-sidecar.rs (hunk 1 above)
Comment (jgrillo-grapl) [reply src/rust/plugin-execution-sidecar/analyzer-execution-sidecar.rs]: fixed

**Comment by jgrillo-grapl** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

@colin-grapl @wimax-grapl just to confirm, have these changes been merged into your branches? If so I can close this PR

**Pull request merged**